File: bannerlayout/src/main/java/com/wkz/bannerlayout/annotation/TipsLayoutSiteMode.java
package com.wkz.bannerlayout.annotation;

import android.support.annotation.IntDef;
import android.widget.RelativeLayout;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@IntDef({
        TipsLayoutSiteMode.LEFT,
        TipsLayoutSiteMode.TOP,
        TipsLayoutSiteMode.RIGHT,
        TipsLayoutSiteMode.BOTTOM,
        TipsLayoutSiteMode.CENTER
})
@Retention(RetentionPolicy.CLASS)
public @interface TipsLayoutSiteMode {
    int LEFT = RelativeLayout.ALIGN_PARENT_LEFT;
    int TOP = RelativeLayout.ALIGN_PARENT_TOP;
    int RIGHT = RelativeLayout.ALIGN_PARENT_RIGHT;
    int BOTTOM = RelativeLayout.ALIGN_PARENT_BOTTOM;
    int CENTER = RelativeLayout.CENTER_IN_PARENT;
}

File: bannerlayout/src/main/java/com/fphoenixcorneae/bannerlayout/annotation/TipsLayoutSiteMode.kt
package com.fphoenixcorneae.bannerlayout.annotation

import android.widget.RelativeLayout
import androidx.annotation.IntDef

@IntDef(
        TipsLayoutSiteMode.LEFT,
        TipsLayoutSiteMode.TOP,
        TipsLayoutSiteMode.RIGHT,
        TipsLayoutSiteMode.BOTTOM,
        TipsLayoutSiteMode.CENTER
)
@kotlin.annotation.Retention(AnnotationRetention.BINARY)
annotation class TipsLayoutSiteMode {
    companion object {
        const val LEFT = RelativeLayout.ALIGN_PARENT_LEFT
        const val TOP = RelativeLayout.ALIGN_PARENT_TOP
        const val RIGHT = RelativeLayout.ALIGN_PARENT_RIGHT
        const val BOTTOM = RelativeLayout.ALIGN_PARENT_BOTTOM
        const val CENTER = RelativeLayout.CENTER_IN_PARENT
    }
}